spec/components/previews/complete/icon_component_preview.rb

```diff
@@ -1,7 +1,26 @@
 module Complete
   class IconComponentPreview < ViewComponent::Preview
-    def default
+    # @!group
+    def uncompleted
       render(IconComponent.new(lesson: Lesson.first, current_user: User.first))
     end
+
+    def completed
+      lesson = Lesson.first
+      current_user = User.last
+      complete_lesson!(lesson, current_user)
+      render(IconComponent.new(lesson:, current_user:))
+    end
+
+    private
+
+    def complete_lesson!(lesson, user)
+      user.lesson_completions.create(
+        lesson_id: lesson.id,
+        lesson_identifier_uuid: lesson.identifier_uuid,
+        course_id: lesson.course.id,
+        path_id: lesson.course.path.id,
+      )
```

Comment:
This lesson copmletion logic is repeated several times at the moment, but it's fairly low impact. I would prefer to figure out a more robust way of adding state when needed (perhaps via the `ApplicationPreviewComponent`) instead of spending too much time abstracting it for this relatively small use case